packages/auth/src/shared/utils.ts

```diff
@@ -7,3 +10,28 @@ export function generateNonce(): string {
 export function generateToken(): string {
   return crypto.randomBytes(10).toString('hex').toUpperCase()
 }
+
+export function verifyApiSignature(ctx: Context, config: IAppConfig): boolean {
+  const { headers, body } = ctx.request
+  const signature = headers['signature']
+  if (!signature) {
+    return false
+  }
+
+  const signatureParts = (signature as string)?.split(', ')
+  const timestamp = signatureParts[0].split('=')[1]
```

Comment:
Replay attack? So nothing here prevents us to get the same and same request over and over again, correct. As long that once it was correct, we can get it n number of times.

I would do it like this:
1. Save the signature to redis for example with key: `signature:${headers['signature']}`
2. On line 20 check if this key exists. If exists, return Unauthorized
3. Check the validity of request ... basically...how long after the request has been signed, we can still process this request? Let's say 30 seconds? Or even less? Also note, that this value is also the TTL of the above signature key

Reply:
Need to do this on the `backend` package as well. Going to capture in https://github.com/interledger/rafiki/issues/2725